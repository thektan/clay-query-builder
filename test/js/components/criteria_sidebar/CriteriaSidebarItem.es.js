import React from 'react';
import CriteriaSidebarItem from 'components/criteria_sidebar/CriteriaSidebarItem.es';
import {cleanup, render} from 'react-testing-library';

const connectDnd = jest.fn(el => el);

describe(
	'CriteriaSidebarItem',
	() => {
		afterEach(cleanup);

		it(
			'should render',
			() => {
				const OriginalCriteriaSidebarItem = CriteriaSidebarItem.DecoratedComponent;

				const {asFragment} = render(
					<OriginalCriteriaSidebarItem
						connectDragSource={connectDnd}
					/>
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);
	}
);