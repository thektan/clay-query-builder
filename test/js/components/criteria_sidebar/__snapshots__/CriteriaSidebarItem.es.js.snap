// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaSidebarItem should render 1`] = `
<DocumentFragment>
  <li
    class="criteria-sidebar-item-root"
  >
    <span
      class="inline-item"
    >
      <svg
        aria-hidden="true"
        class="lexicon-icon lexicon-icon-drag"
        viewBox="0 0 512 512"
      >
        <use
          xlink:href="#drag"
        />
      </svg>
    </span>
    <span
      class="criteria-sidebar-item-type sticker sticker-light"
    >
      <span
        class="inline-item"
      >
        <svg
          aria-hidden="true"
          class="lexicon-icon lexicon-icon-text"
          viewBox="0 0 512 512"
        >
          <use
            xlink:href="#text"
          />
        </svg>
      </span>
    </span>
  </li>
</DocumentFragment>
`;
